pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -578,3 +582,292 @@ func TestFlowAggregator_closeUpdateChBeforeFlowExportLoopReturns(t *testing.T) {
+	want := querier.Metrics{
+		NumRecordsExported: 1,
+		NumRecordsReceived: 1,
+		NumFlows:           1,
+		NumConnToCollector: 1,
+	}
+
+	t.Run("non-empty metrics", func(t *testing.T) {
+		fa := &flowAggregator{
+			collectingProcess:  mockCollectingProcess,
+			aggregationProcess: mockAggregationProcess,
+			numRecordsExported: 1,
+		}
+
+		mockCollectingProcess.EXPECT().GetNumRecordsReceived().Return(int64(1))
+		mockAggregationProcess.EXPECT().GetNumFlows().Return(int64(1))
+		mockCollectingProcess.EXPECT().GetNumConnToCollector().Return(int64(1))
+
+		got := fa.GetRecordMetrics()
+		assert.Equal(t, want, got)
+	})
+}
+
+func TestFlowAggregator_InitCollectingProcess(t *testing.T) {
+	tests := []struct {
+		name                        string
+		aggregatorTransportProtocol flowaggregatorconfig.AggregatorTransportProtocol
+		flowAggregatorAddress       string
+		k8sClient                   kubernetes.Interface
+		expectedErr                 string
+	}{
+		{
+			name:                        "TLS protocol",
+			aggregatorTransportProtocol: flowaggregatorconfig.AggregatorTransportProtocolTLS,
+			k8sClient:                   fake.NewSimpleClientset(),
+			flowAggregatorAddress:       "192.168.1.2",
+		},
+		{
+			name:                        "TCP protocol",
+			aggregatorTransportProtocol: flowaggregatorconfig.AggregatorTransportProtocolTCP,
+			k8sClient:                   fake.NewSimpleClientset(),
+		},
+		{
+			name:                        "TLS protocol with empty flowAggregatorAddress",
+			aggregatorTransportProtocol: flowaggregatorconfig.AggregatorTransportProtocolTLS,
+			k8sClient:                   fake.NewSimpleClientset(),
+			expectedErr:                 "error when creating server certificate: lookup : no such host",
+		},
+		{
+			name:      "neither TLS nor TCP protocol",
+			k8sClient: fake.NewSimpleClientset(),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fa := &flowAggregator{
+				aggregatorTransportProtocol: tt.aggregatorTransportProtocol,
+				flowAggregatorAddress:       tt.flowAggregatorAddress,
+				k8sClient:                   tt.k8sClient,
+			}
+			err := fa.InitCollectingProcess()
+			if tt.expectedErr != "" {
+				assert.EqualError(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestFlowAggregator_InitAggregationProcess(t *testing.T) {
+
+	fa := &flowAggregator{
+		activeFlowRecordTimeout:     testActiveTimeout,
+		inactiveFlowRecordTimeout:   testInactiveTimeout,
+		aggregatorTransportProtocol: flowaggregatorconfig.AggregatorTransportProtocolTCP,
+	}
+	err := fa.InitCollectingProcess()
+	require.NoError(t, err)
+
+	err = fa.InitAggregationProcess()
+	require.NoError(t, err)
+}
+
+func TestFlowAggregator_fillK8sMetadata(t *testing.T) {
+	dummySrcPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "dummySourcePod",
+		},
+		Spec: v1.PodSpec{
+			NodeName: "dummySourceNode",
```

Comment:
"dummy" can be considered non-inclusive / offensive. I think the "dummy" prefix is not necessary here, and you could remove it

Reply:
removed prefix "dummy"